Bug 1477353

fbertsch opened the pull request.

chutten commented on mozaggregator/service.py:
```diff
@@ -304,10 +309,19 @@ def get_dates_metrics(prefix, channel):
+
+    if (channel == RELEASE_CHANNEL and
+       not SHOW_RELEASE_METRICS and
+       metric not in PUBLIC_RELEASE_METRICS):
```
So `SHOW_RELEASE_METRICS` means show _all_ release metrics? That wasn't clear to me on reading the name.

fbertsch replied:
Right, bikeshed is fine here, that seems reasonable

robhudson commented on mozaggregator/service.py:
```diff
@@ -304,10 +309,19 @@ def get_dates_metrics(prefix, channel):
+
+    if (channel == RELEASE_CHANNEL and
+       not SHOW_RELEASE_METRICS and
+       metric not in PUBLIC_RELEASE_METRICS):
```
This could be moved to a small function that returns a boolean, like `if _show_public_release_metric(channel, metric)`, and if there's future logic that needs to be added it's easy to add it and easy to test it.

chutten replied:
Oooh, I like that. The logic may even be easier to read if it is in early-return format

```python
if SHOW_ALL_RELEASE_METRICS:
    return true
if metric in PUBLIC_RELEASE_METRICS:
    return true
return false
```

robhudson commented on mozaggregator/service.py:
```diff
@@ -58,6 +58,11 @@
 
+# Allowed public release metrics
+RELEASE_CHANNEL = "release"
+SHOW_RELEASE_METRICS = (os.environ.get("SHOW_RELEASE_METRICS", "False") == "True")
```
The outer parens aren't really necessary here.

fbertsch commented:
@robhudson @chutten thanks for the feedback! I've moved the boolean to it's own function, and in addition I split up the `METRICS_BLACKLIST`: we now can blacklist a metric totally, or just disable it for public viewing. I've also updated the README.

robhudson commented:
:+1: Looks great!

chutten commented on README.md:
```diff
@@ -37,6 +37,19 @@ make shell
+To enable all release metrics (except those in `METRICS_BLACKLIST`), set the envvar `ALLOW_ALL_RELEASE_METRICS` to "True".
+
+### Pre-Release Metrics
+To disable a metric for prerelease only (in effect, allowing it for viewing internally only), add it to `PRERELEASE_METRICS_BLACKLIST`.
```
I'd drop the parenthetical remark as there's nothing internal/external about the service yet.

chutten commented on mozaggregator/service.py:
```diff
@@ -57,12 +57,17 @@
+# Allowed public release metrics
+RELEASE_CHANNEL = "release"
+ALLOW_ALL_RELEASE_METRICS = os.environ.get("ALLOW_ALL_RELEASE_METRICS", "False") == "True"
+PUBLIC_RELEASE_METRICS = {"SCALARS_TELEMETRY.TEST.KEYED_UNSIGNED_INT"}
```
If we expect this to be large, should we pop this out to a resource?
Scalars are grouped by category, would it make sense to support globs/regexes so we could support SCALARS_TELEMETRY.TEST.*?

fbertsch replied:
Let's handle these cases in the future, depending on the outcome of the project. Certainly we could change it down the line.

chutten commented on mozaggregator/service.py:
```diff
@@ -309,10 +331,10 @@ def get_dates_metrics(prefix, channel):
-    if metric in METRICS_BLACKLIST:
-        abort(404)
+    if not _allow_metric(channel, metric):
+        abort(404, description="This metric is not allowed.")
```
Are these messages to be useful to external developers as well as us? If so we may wish to elaborate on what "not allowed" means. "not allowed to be served" or something.

fbertsch commented:
@chutten I agree, it does seem to be more about whether we're authed or not. In that case we can back out the blacklist splitting piece from here, and add it back in again when we add the auth.
